vapi/compute/policy.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright (c) 2019-2020 VMware, Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0.
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compute
+
+import (
+	"context"
+	"net/http"
+	"path"
+
+	"github.com/vmware/govmomi/vapi/compute/internal"
+	"github.com/vmware/govmomi/vapi/rest"
+)
+
+// PolicyManager provides convenience methods to operate Compute Policy
+type PolicyManager struct {
+	*rest.Client
+}
+
+// NewPolicyManager creates a new PolicyManager with the given client
+func NewPolicyManager(client *rest.Client) *PolicyManager {
+	return &PolicyManager{
+		Client: client,
+	}
+}
+
+// Policy represents the structure a compute policy.
+type Policy struct {
+	Capability  string `json:"capability,omitempty"`
+	Description string `json:"description,omitempty"`
+	Name        string `json:"name,omitempty"`
+	Policy      string `json:"policy,omitempty"`
+}
+
+// PolicySpec represents the structure to create a compute policy.
+type PolicySpec struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	Capability  string `json:"capability,omitempty"`
+	HostTag     string `json:"host_tag,omitempty"`
+	VMTag       string `json:"vm_tag,omitempty"`
+}
```

Comment:
Do we need 2 structs? Due to the use of omitempty, string Policy @ line 45 can be added to PolicySpec?

Reply:
It could be yes, but 'PolicySpec' has more inputs (HostTag, VMTag, etc.) that will never be included as output with 'List'.  So while it'd be functional using a single type, it can be confusing that way.  While they have fields in common, the REST API documents them as different types.

Reply:
Actually, 'Get' does return HostTag, VMTag, etc., its just 'List' that doesn't return those. I'll take another look, maybe it would be simpler to collapse into a single type as we had it originally.

Reply:
@shylasrinivas ok made this change and the others we discussed, can you review again?

Reply:
lgtm, thanks!